torch/_dynamo/guards.py

```diff
@@ -359,6 +359,15 @@ def HASATTR(self, guard: Guard):
 
         self._produce_guard_code(guard, [code], provided_guarded_object=self.get(base))
 
+    def VMAP_INCREMENT_NESTING_MATCH(self, guard: Guard):
+        # Invalidate the graph if a call to vmap has been made prior to this
+        # This is super conservative as the interpreter stack may not contain
+        # vmap
+        code = [
+            "torch._C._functorch.peek_interpreter_stack() is None",
+        ]
+        self._produce_guard_code(guard, code)
```

Comment:
Should the guard be a check on the current level index instead of if the interpreter stack is None?

If so, then we should change torch._C._functorch.current_level to return Optional[int] and change the guard to be a FUNCTORCH_CURRENT_LEVEL_MATCH